viewing_party/party.py

```diff
@@ -1,23 +1,145 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie = {}
 
+    if title and genre and rating:
+        movie["title"] = title
+        movie["genre"] = genre
+        movie["rating"] = rating
+        return movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+    watched_list = user_data["watched"]
+    watched_list.append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = user_data["watchlist"]
+    watchlist.append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    counter = 0
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == movie:
+            user_data["watched"].append(movie_dict)
+            user_data["watchlist"].pop(counter)
+        counter += 1
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_sum = 0.0
+    average = 0.0
+
+    for movie_dict in user_data["watched"]:
+        rating_sum += movie_dict["rating"]
+    
+    if user_data["watched"]:
+        average = rating_sum / len(user_data["watched"])
+        return average
+    else:
+        return average

```

Comment:
Looks good! Another way you could write this to return average == 0.0 when "watched" is empty is to write it in an if statement like a guard clause at the beginning of this function:

```
if not user_data["watched"]:
      return 0.0
```

This also makes your intent more explicit ("if watched is Falsy then the average is 0.0) versus having it in an else block which can require someone else reading your code to have to step through the entire logical flow to figure out what's going on. 

If you follow that approach then you could get rid of average = 0.0 on line 37